File: veloProWeb/src/main/java/com/veloProWeb/Model/Entity/Dispatch.java
package com.veloProWeb.Model.Entity;

import com.veloProWeb.Model.Entity.Sale.Sale;
import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Entity
@Data
@Table(name = "dispatches")
public class Dispatch {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String trackingNumber;
    private String status;

    @DateTimeFormat(pattern = "dd-MM-yyyy")
    @Temporal(TemporalType.DATE)
    @CreatedDate
    private LocalDate created;
    @DateTimeFormat(pattern = "dd-MM-yyyy")
    @Temporal(TemporalType.DATE)
    @CreatedDate
    private LocalDate deliveryDate;

    @ManyToOne
    @JoinColumn(name = "sale_id", nullable = false)
    private Sale sale;
}

File: veloProweb-frontend/src/app/models/Entity/Sale/dispatch.ts
import { SaleDetailDTO } from "../../DTO/sale-detail-dto";

export interface Dispatch {
    id: number,
    trackingNumber: string,
    status: string,
    address: string,
    comment: string,
    customer: string,
    created: string,
    deliveryDate: string,
    saleDetail: SaleDetailDTO | null
}
